docs/pull_request_template.md

```diff
@@ -0,0 +1,17 @@
+- [ ] Link to pivotal/JIRA issue
```

Comment:
I was expecting these to be headers (like `##`) instead of checkboxes, and then the person filling this our just adds their thoughts under it, what do you think?

Reply:
Yep, I suppose headers make a little more sense.